zng/recorditer.go

```diff
@@ -0,0 +1,97 @@
+package zng
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/brimsec/zq/zcode"
+)
+
+var (
+	ErrExhausted = errors.New("called Next() on iterator after last record")
+	ErrMismatch  = errors.New("mismatch between record type and value")
+)
+
+type iterInfo struct {
+	iter     zcode.Iter
+	typ      *TypeRecord
+	offset   int
+	fullname string
+}
+
+type recordIter struct {
+	stack []iterInfo
+}
+
+func (r *Record) NewFieldIterator() recordIter {
```

Comment:
Nits: We usually a type's methods in the same file, so put this in recordval.go. And call it call it FieldIter so it agrees with (*Record).ZvalIter.

Reply:
nit: should recordIter be called fieldIter (or vice versa)?